doc/03-omnifest/02-external.md

```diff
@@ -1,22 +1,117 @@
 # External
 
-## osbuild
+External directives are directives that are implemented externally from `otk`.
+They are meant to be used to provide target-specific behavior and can be used
+to express things where `otk` is not expressive enough.
+
+## Protocol
+
+Directives are small executables that receive JSON and are expected to output
+JSON again in a specific format. Additional keys are not allowed.
+
+When we have an [omnifest](./index.md) that looks like this:
+
+```yaml
+otk.external.name:
+  child:
+    - 1
+  options: "are here"
+```
+
+The external gets called with the following JSON:
+
+```json
+{
+  "tree": {
+    "otk.external.name": {
```

Comment:
Curious, why do we include the name in the tree? In case an external is symlinked and would change "personality" based on the inputs? But even then it would know this from argv0, no?

Reply:
It's to make it a bit more explicit to show which part of the tree it is operating on.

Reply:
Fwiw, the spec and the (current) implementation diverge here, when adding the following test:
```python
def test_integration_happy(tmp_path):
    os.environ["OTK_EXTERNAL_PATH"] = os.fspath(tmp_path)
    fake_external_path = tmp_path / "otk_external_test"
    fake_external_path.write_text(textwrap.dedent("""\
    #!/bin/sh
    cat - > "$0".stdin
    echo '{"tree": {"some": "result"}}'
    """))
    os.chmod(fake_external_path, 0o755)

    fake_directive = "otk.external.test"
    fake_tree = {
        "foo": "bar",
        "sub": {
            "baz": "foobar",
        },
    }

    res = otk.external.call(fake_directive, fake_tree)
    assert res == {"some": "result"}

    inp = fake_external_path.with_suffix(".stdin").read_text()
    assert json.loads(inp) == {"tree": fake_tree}
```
(note the last two lines that are new compared to the patch I attached to this originally).

Then it shows that the `otk.external.name` is (currently) not part of the inputs. Not a big deal but I wanted to mention it so that we are aware :)